pkg/deploy/gateway/gateway.go

```diff
@@ -56,8 +58,9 @@ var (
 
 // SyncGatewayToCluster installs or deletes the gateway based on the custom resource configuration
 func SyncGatewayToCluster(deployContext *deploy.DeployContext) error {
-	if util.GetServerExposureStrategy(deployContext.CheCluster) == "single-host" &&
-		(deploy.GetSingleHostExposureType(deployContext.CheCluster) == "gateway") {
+	if (util.GetServerExposureStrategy(deployContext.CheCluster) == "single-host" &&
```

Comment:
`a && b || c` let's be more clear here: `(a && b) || c`

Reply:
I don't follow, it's now written as `(a && b) || c` => `(single-host && gateway) || devworkspace`